### ISSUE TITLE
Overriding encoding

### ISSUE THREAD

**Issue by danielcrane** [daijro/hrequests#31]
In the `requests` library, if the wrong type of encoding is used then it's possible to manually fix this by overriding the `request.encoding` attribute.

In `hrequests`, this is unfortunately not possible, as doing so leads to the following exception:
```python
request.encoding = 'euc_kr'
AttributeError: property 'encoding' of 'Response' object has no setter
```

I was wondering if there's any other way to achieve this manual override, as some websites I'm working with are being encoded incorrectly.

**Comment by Cosmicoppai** [comment]
@danielcrane I think It has probably to do with https://github.com/bogdanfinn/tls-client/blob/e09a70ebbd208feab47f2bf05a10bdd74be5f142/cffi_src/factory.go#L213C30-L213C30

**Comment by Cosmicoppai** [comment]
@daijro Please consider including both the content in bytes and text in the response? This would provide added convenience for decoding using different encoding systems.

**Comment by danielcrane** [comment]
Ah, it's not ideal, but temporarily it's possible to make use of the `tls-client`'s [readAllBodyWithStreamToFile](https://github.com/bogdanfinn/tls-client/blob/e09a70ebbd208feab47f2bf05a10bdd74be5f142/cffi_src/factory.go#L202) function by overriding the `request_payload` here and adding the `StreamOutputPath` key:

https://github.com/daijro/hrequests/blob/4953babe2fb76412629e08967ea19575e83ba97e/hrequests/client.py#L337-L357

This'll result in an intermediate file containing the raw `bytes` data being outputted to the specified location.

If modifying the Golang code would be too much of a pain, an alternative could be using the `tempfile` module to create a temporary file to output the bytes data to, and then reading from that file and returning it in `requests.content`.

**Comment by danielcrane** [comment]
Submitted a draft PR that outlines a purely Python-based fix to this issue, it's not very elegant but not got any brighter ideas at this point.

**Comment by danielcrane** [comment]
@Cosmicoppai of course, yes.

**Comment by daijro** [comment]
Hello, this should be fixed in v0.8.0-beta.2.

I was able to get the [`respBodyBytes`](https://github.com/bogdanfinn/tls-client/blob/e09a70ebbd208feab47f2bf05a10bdd74be5f142/cffi_src/factory.go#L213C30-L213C30) object converting to a string by reimplementing the [BuiltResponse](https://github.com/daijro/hrequests/blob/main/bridge/server.go#L372) function to encode the data in Base64 if the data was non UTF-8. From Python, I can now decode the bytes into any type of encoding.

`Response.encoding` should now be automatically calculated (using chardet), `Response.content` always returns a `bytes` object, and you can also change `Response.encoding` to adjust the encoding.

Sorry for the late reply, I've been focused on schoolwork lately. Let me know if the change works for you!
